src/Domain/FunctionsOfTime/QuaternionHelpers.cpp

```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/FunctionsOfTime/QuaternionHelpers.hpp"
+
+#include <boost/math/quaternion.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
```

Comment:
You don't need to include things that are included in the header.  The gsl include can be replaced with a forward-declaration in the header, however.

Reply:
@wthrowe Your comment confuses me. Haven't we operated under the "include-what-you-use" guideline to include (or forward-declare) everything used in the file?

Reply:
I don't know which you prefer, but originally I had @wthrowe's method.

Reply:
The hpp counts as part of the cpp for iwyu purposes.  However, I just tested and the include-what-you-use program doesn't suggest removing them if they are there, and also doesn't suggest adding them if they are not, so I'll consider either way acceptable.

In any case, though, the Gsl include in the header should change to a forward declaration of not_null.

Reply:
Why a forward declaration here and not an include?

Reply:
In general, forward declarations are better because they require the compiler to parse less code, and so can speed up compiling.  In this specific case it is allowed because function arguments only have to be complete types if the function is defined or called (and anyone calling the function will have to have included `Gsl.hpp` in order to construct the `gsl::not_null` to pass in, so the argument will be complete by then).